src/containers/ProfilePage/index.jsx

```diff
@@ -1,34 +1,91 @@
 import { Container, Row, Accordion } from "react-bootstrap";
 import { useForm } from "react-hook-form";
+import Joi from "joi";
+import { joiResolver } from "@hookform/resolvers/joi";
 import UserDetailForm from "../../components/profile/UserDetailForm";
 import UserEmailForm from "../../components/profile/UserEmailForm";
 import UserPasswordForm from "../../components/profile/UserPasswordForm";
+import { AuthContext } from "../../auth/Authentication";
 import defaultProfileImage from "../../images/defaultProfileImage.png";
 import ProfileSection from "./ProfileSection";
+import { validationSchemaObject } from "../../utils/formHelpers";
 import "./index.scss";
 
-export default function ProfilePage({ submit }) {
-  const { register, handleSubmit, errors } = useForm();
+const profileFormSchema = Joi.object({
+  address: validationSchemaObject.anyOptional,
+  birthdate: validationSchemaObject.anyOptional,
+  city: validationSchemaObject.anyOptional,
+  countryCode: validationSchemaObject.anyOptional,
+  firstName: validationSchemaObject.anyOptional,
+  lastName: validationSchemaObject.anyOptional,
+  nationality: validationSchemaObject.anyOptional,
+  phoneNumber: validationSchemaObject.anyOptional,
+  sex: validationSchemaObject.anyOptional,
+  state: validationSchemaObject.anyOptional,
+  zipCode: validationSchemaObject.anyOptional,
+  email: validationSchemaObject.email,
+});
+
+const EMAIL_UPDATED = "Email updated successfully!";
+
+export default function ProfilePage() {
+  const [updateStatus, setUpdateStatus] = React.useState(null);
+  const currentUser = React.useContext(AuthContext);
+
+  const { register, handleSubmit, errors } = useForm({
+    mode: "onTouched",
+    defaultValues: {
+      email: currentUser?.email,
+    },
+    resolver: joiResolver(profileFormSchema),
+  });
 
   const sections = [
     {
       label: "User details",
-      children: <UserDetailForm register={register} errors={errors} />,
+      children: (
+        <UserDetailForm
+          register={register}
+          errors={errors}
+          currentUser={currentUser}
+        />
+      ),
     },
     {
       label: "Email address",
-      children: <UserEmailForm register={register} errors={errors} />,
+      children: (
+        <UserEmailForm
+          register={register}
+          errors={errors}
+          currentUser={currentUser}
+        />
+      ),
     },
     {
       label: "Change password",
-      children: <UserPasswordForm register={register} errors={errors} />,
+      children: (
+        <UserPasswordForm
+          register={register}
+          errors={errors}
+          currentUser={currentUser}
+        />
+      ),
     },
   ];
 
   const onSave = (data) => {
-    // TODO: save data in the database
-    submit(data);
+    // check if email changed and update it
+    if (data.email !== currentUser.email) {
+      currentUser
+        .updateEmail(data.email)
+        .then(function () {
+          setUpdateStatus(EMAIL_UPDATED);
```

Comment:
email updated what. is it a message?

Reply:
yeah, to be shown as status when the user click on the "Save changes" button.